### PR TITLE
Add Create Task API quality assurance table

### LINKED ISSUE #472: Create Quality Assurance table for Create Task API
## Description

As a developer,
I need to know what are the test cases for a certain API,
so that I can write new unit tests.

As a tester,
I need to know what are the test cases for a certain API,
so that I can perform manual tests on the API.

The task is to come up and add a quality assurance table to the docs in the `/docs` folder.

### Update [Required]
- [ ] Add test cases for Create Task API

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by AjayThakur12**
### Description
Create d Quality Assurance table for Create Task API in docs/quality-assurance-test-cases.md

Fixes #472 

### Type of Change:
- Documentation

### How Has This Been Tested?
I checked both android - mentorship code services code and backend API code

### Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have made corresponding changes to the documentation
- [ ] Update Postman API at /docs folder

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by ramitsawhney27 — DISMISSED** [review]
Can we keep consistent spacing?

**Comment by AjayThakur12** [comment]
@ramitsawhney27  this is how it look after adding not able to understand your point
yui

File: docs/quality-assurance-test-cases.md [hunk 1]
```diff
@@ -172,3 +172,13 @@ Only admin users have access to this.
...
 | Revoking an admin user, when the current user is not an admin | Fail |
 
+## Tasks
+
+### Create
+**Service:** POST /mentorship_relation/{relation_id}/task
+
+| Test Case       | Outcome |
+| ------------- | ------------- |
+|Create a task if there is a valid relation exist between user and mentor and relation state is accepted | Success |
```
Comment (ramitsawhney27) [inline docs/quality-assurance-test-cases.md]: I meant from a source markdown perspective as well. Look at this line

File: docs/quality-assurance-test-cases.md [hunk 2]
```diff
@@ -172,3 +172,13 @@ Only admin users have access to this.
...
+| Create a task if there is no relation between user and mentor | Fail |
```
Comment (isabelcosta) [inline docs/quality-assurance-test-cases.md]: I am not sure if anyone could even test this, can you be more specific? I think what you want to say here is something like:
create a task when logged user is not involved in the relation

File: docs/quality-assurance-test-cases.md [hunk 3]
```diff
@@ -172,3 +172,13 @@ Only admin users have access to this.
...
+| Create a task if relation state is not accepted | Fail |
```
Comment (isabelcosta) [inline docs/quality-assurance-test-cases.md]:
```suggestion
| Create a task if relation state is different than accepted | Fail |
```
its more clear like this, I think

File: docs/quality-assurance-test-cases.md (hunk 1 above)
Comment (isabelcosta) [inline docs/quality-assurance-test-cases.md]:
```suggestion
| Create a task for a relation, in the accepted state, between logged user and another user | Success |
```

**Review by isabelcosta — CHANGES_REQUESTED** [review]
Thank you for contributing to our documentation @AjayThakur12 🎉 
I left some notes to make the text a little more clear.
Here is one more case for you to add:
- creating a task without a description (either empty or not in the request body at all)

**Comment by isabelcosta** [comment]
Because of this PR I detected a bug on the backend :D I am happy, because it means this is useful!
https://github.com/anitab-org/mentorship-backend/issues/507

**Comment by AjayThakur12** [comment]
@isabelcosta  thank you for your review I will make required changes

**Comment by isabelcosta** [comment]
@AjayThakur12 you also have to fix merge conflicts. Do you need any help with this PR? Asking as there are no updates

**Comment by AjayThakur12** [comment]
@isabelcosta  sorry for the delay, I  will complete  task by today.

**Comment by developer22-university** [comment]
@isabelcosta  is this resolved? i can help if required

**Review by isabelcosta — APPROVED** [review]
Looks good @AjayThakur12 :) thank you for your contribution 🎉

**Comment by isabelcosta** [comment]
@AjayThakur12 great work 👏 your PR was merged!

**Pull request merged**